AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet2Solutions.lean

```diff
@@ -0,0 +1,283 @@
+/-
+This is a solutions sheet.
+-/
+
+import Mathlib.Tactic
+
+/-Credit for some exercises goes to Kevin Buzzard and his Formalizing Mathematics course OR to the
+Natural Number Game. If you would like to learn more about Lean, Buzzard's course goes more in depth
+in relation to numerous undergraduate math topics. When exercises are from either of these sources,
+they will be clearly marked so as to give credit.
+
+Formalising Mathematics can be found here:
+<https://www.ma.imperial.ac.uk/~buzzard/xena/formalising-mathematics-2024/index.html>-/
+
+/-
+Many basic tactics in Lean are best introduced through logic exercises. You already know "rfl" and
+"rw" from the Natural Number Game (NNG), and you are also already familiar with a few other basic
+tactics. However, keep in mind that the way these tactics work in the Natural Number Game may be
+slightly different than the way that they are actually used in Lean. These differences mostly boil
+down to slight discrepancies in syntax.
+
+Mathlib is the library of theorems that have already been formalized in Lean. Theorems in mathlib
+are named and can be used in other proofs. An overview of the topics currently in mathlib can be
+found here: <https://leanprover-community.github.io/mathlib-overview.html> Note that "#check" tells
+the type. For propositions, this will tell you the definition of a proposition in mathlib. For
+example:
+-/
+
+#check zero_add
+
+/-
+Let's start with a few tactics that you are already familiar with from the NNG:
+
+exact
+apply
+intro
+
+To continue getting used to reading Lean and working with the Lean Infoview, let's do a few levels
+from the NNG. Don't forget that you can also use "rw" and "rfl"! Delete the "sorry"s and fill in the
+proofs.
+-/
+
+/-NNG Implication World Level 1-/
+example (x y z : Nat) (h1 : x + y = 37) (h2 : 3 * x + z = 42) : x + y = 37 := by
```

Comment:
Good point, but I didn't do this so that the student would be able to see x y z defined clearly (and since I put in a little note about `variable` a few exercises down)